pkg/apis/azureprovider/v1beta1/types.go

```diff
@@ -401,11 +401,16 @@ type Image struct {
 type VMIdentity string
 
 type OSDisk struct {
-	OSType      string      `json:"osType"`
-	ManagedDisk ManagedDisk `json:"managedDisk"`
-	DiskSizeGB  int32       `json:"diskSizeGB"`
+	OSType      string                `json:"osType"`
+	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
```

Comment:
What's the benefit of renaming ManagedDisk -> ManagedDiskParameters?  Clearly it doesn't affect the serialisation but it'll cause a rename for any Go developer who initialises a ManagedDisk{} when they revendor (e.g. see the stubs.go change below).  I'm only seeing downside.

Reply:
I was keeping inline with what is in upstream. 
OSDisk: https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-12-01/compute/models.go#L6700
ManagedDiskParameters: https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-12-01/compute/models.go#L6507

If we prefer to keep our own naming then I can revert.

Reply:
We're not renaming the field name, just the type name, which is not exposed to users and won't require any changes.  +1 to aligning with upstream.